ml.net/InclusiveCodeReviews.LLMtoCSV/Program.cs

```diff
@@ -0,0 +1,98 @@
+﻿// This is a "script" to iterate over all the negative rows in classified.csv
+// It outputs to output.csv in the current directory (bin/Debug/net8.0)
+// We take 10 rows at a time and ask an LLM to "reword" the negative rows
+
+using System.Globalization;
+using System.Text.Json;
+using CsvHelper;
+using CsvHelper.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+// SemanticKernel setup
+var builder = Kernel.CreateBuilder();
+builder.Services.AddLogging(c => c.SetMinimumLevel(LogLevel.Debug).AddDebug());
+builder.Services.AddOpenAIChatCompletion(modelId: "gpt-4", apiKey: "FILL_ME_OUT");
+
+Kernel kernel = builder.Build();
+
+const string FunctionDefinition =
+	"""
+	{{sentences}}
+
+	Answer to the following questions using JSON syntax, using the format: { "comments": ["comment1", "comment2", ...] }
+	You are expert software engineer that is particularly good at writing inclusive, well-written, thoughtful code reviews.
+	Reword each comment in the JSON above to be more inclusive, constructive, and nice.
+	@github is a placeholder for a GitHub username, include this in each comment as-is.
+	#code and #url are placeholders for code and URLs, include this in each comment as-is.
+	""";
+
+var plugin = kernel.ImportPluginFromType<BatchedCommentPlugin>("Comments");
+var promptTemplateFactory = new KernelPromptTemplateFactory();
+var promptTemplate = promptTemplateFactory.Create(new PromptTemplateConfig(FunctionDefinition));
+var renderedPrompt = await promptTemplate.RenderAsync(kernel);
+
+// Create the "prompt" function
+var function = kernel.CreateFunctionFromPrompt(FunctionDefinition, new OpenAIPromptExecutionSettings { MaxTokens = 1000 });
+
+// CSV setup
+string location = Path.GetDirectoryName(typeof(Program).Assembly.Location)!;
+var path = Path.Combine(location, "..", "..", "..", "..", "..", "comments", "classified.csv");
+var configuration = new CsvConfiguration(CultureInfo.InvariantCulture)
+{
+	HasHeaderRecord = true,
+	NewLine = Environment.NewLine,
+};
+using var reader = new StreamReader(path);
+using var inputCSV = new CsvReader(reader, configuration);
+using var writer = File.CreateText("output.csv");
+using var csvWriter = new CsvWriter(writer, configuration);
+
+foreach (var comment in inputCSV.GetRecords<Comment>())
```

Comment:
maybe use the async overload here? https://github.com/JoshClose/CsvHelper/blob/8599e80f2a40fe073eeb5a281348169db7641be9/src/CsvHelper/IReader.cs#L88